PyInstaller/loader/pyiboot01_bootstrap.py

```diff
@@ -47,12 +47,24 @@
 # Some packages behaves differently when running inside virtual environment.
 # E.g. IPython tries to append path VIRTUAL_ENV to sys.path.
 # For the frozen app we want to prevent this behavior.
-VIRTENV = 'VIRTUAL_ENV'
-if VIRTENV in os.environ:
+#
+# But, even though we have enough reason to delete VIRTUAL_ENV in environment,
+# we should still keep the original data for those who relies on it essentially,
+# provide another way to access it when necessary.
+preserved_env_prefix = 'PYINSTALLER_PRESERVED_'  # could be used for other preserved env in the future
+env_virtual_env = 'VIRTUAL_ENV'
+preserved_env_virtual_env = preserved_env_prefix + env_virtual_env
+
+# NOTE the best way to decide which environment variables to preserve is by
+# passing an option `--preserve-env` to pyinstaller or spec file, then store
+# the preserved ones in the bundle app, get them here to loop and handle,
+# this is intended to be implemented in this feature branch.
+if env_virtual_env in os.environ:
```

Comment:
Please keep the old variable name (VIRTENV) to keep number of chances small.

Also setting `preserved_env_prefix` should go into the `if`-clause to avoid latency (even if it is small).

Reply:
Fixed, put `preserved_env_prefix + VIRTENV` in if block, but `preserved_env_prefix` is still outside because next I want to implement a more generic way to preserve env vars

Reply:
> because next I want to implement a more generic way to preserve env vars

IC. What about upper-casing `preserved_env_prefix` than?

Reply:
Surely it could be, if you prefer that kind of naming convention :)

Reply:
Just added a fixup commit, will squash it before merge.